src/pudl_archiver/archivers/classes.py

```diff
@@ -286,12 +286,14 @@ def dataset_validate_archive(
         """Hook to add archive validation tests specific to each dataset."""
         return []
 
-    def valid_year(self, year: int | str):
+    def valid_year(self, year: int | str | None):
         """Check if this year is one we are interested in.
 
         Args:
             year: the year to check against our self.only_years
         """
+        if not year:
```

Comment:
Just dealing with some complaints from the typechecker. I suppose I could move the "if not year" logic to the caller, too, but that would entail either pulling that calling code into a separate function or adding a weird conditional inside a list comprehension, which sucks.

Reply:
Is this because sometimes there is no year partition? Not following the logic here of adding `None`